sklearn/utils/_array_api.py

```diff
@@ -86,11 +86,20 @@ def yield_namespace_device_dtype_combinations(include_numpy_namespaces=True):
             yield array_namespace, None, None
 
 
-def _check_array_api_dispatch(array_api_dispatch):
-    """Check that array_api_compat is installed and NumPy version is compatible.
+def _check_array_api_dispatch(array_api_dispatch, strict=False):
+    """Check that required dependencies are installed in new enough versions.
 
-    array_api_compat follows NEP29, which has a higher minimum NumPy version than
-    scikit-learn.
+    We need the array_api_compat package as well as new enough versions of
+    NumPy and SciPy.
+
+    Parameters
+    ----------
+    array_api_dispatch : bool
+        Enable or disable array API checks.
+
+    strict : bool, default=False
```

Comment:
Maybe `strict=False/True` could be renamed to something more explicit such as `misconfigured_scipy="warn"` vs `misconfigured_scipy="raise"`.

Reply:
Good point. I went with `strict` thinking that maybe in the future we want to make other checks strict as well. You think it is a YAGNI?

Reply:
I think so. Since it's a private function, we can rename to a more generic name in the future if needed.